src/services/Outbox.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import { encodeArrayBuffer } from '../helpers/base64.js'
+import { logger } from '../helpers/logger.js'
+
+type Sendable = {
+	steps: string[], awareness: string
+}
+
+export default class Outbox {
+	#awarenessUpdate = ''
+	#syncUpdate = ''
+	#syncQuery = ''
+
+	storeStep(step: Uint8Array) {
+		const encoded = encodeArrayBuffer(step)
+		if (encoded < 'AAA') {
```

Comment:
This condition seems a bit superfluous to me given that we log the same message at the end of the function if no other condition matched.

Reply:
But then we'd always need to check if encoded is bigger 'AAA' in the other conditions.

We could have one condition in the beginning that checks if it's valid (i.e. > AAA and < 'Ag') and then do the rest.

Reply:
I guess the size comparison is very fast and don't know if this is really relevant from a performance perspective. Maybe it's also relevant how often we actually get messages that are smaller than `AAA` (and bigger than `Ag`) in practice. In any case no blocker, fine to keep it as is :blush: